### PR TITLE
hibernate-relations Solution

### PR DESCRIPTION
**Pull request by luk14n**

### CODE REVIEW

File: src/main/java/mate/academy/hibernate/relations/Main.java [hunk 1]
```diff
@@ -16,17 +22,20 @@ public static void main(String[] args) {
...
-        ActorService actorService = null; // TODO: initialize this instance
+        ActorService actorService = new ActorServiceImpl(
+                new ActorDaoImpl(sessionFactory)); // TODO: initialize this instance
         actorService.add(vinDiesel);
 
         Movie fastAndFurious = new Movie("Fast and Furious");
         fastAndFurious.setActors(List.of(vinDiesel));
-        MovieService movieService = null; // TODO: initialize this instance
+        MovieService movieService = new MovieServiceImpl(
+                new MovieDaoImpl(sessionFactory)); // TODO: initialize this instance
```
Comment (kerrrusha) [inline src/main/java/mate/academy/hibernate/relations/Main.java:38]: remove TODOs - you've completed them

File: src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java [hunk 2]
```diff
@@ -1,16 +1,27 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.ActorDao;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Actor;
 import mate.academy.hibernate.relations.service.ActorService;
 
 public class ActorServiceImpl implements ActorService {
+
```
Comment (kerrrusha) [inline src/main/java/mate/academy/hibernate/relations/service/impl/ActorServiceImpl.java:9]: redundant blank line
```suggestion
```

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 3]
```diff
@@ -1,16 +1,27 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.exception.DataProcessingException;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
 
 public class CountryServiceImpl implements CountryService {
+
```
Comment (kerrrusha) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:9]:
```suggestion
```

File: src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -12,11 +15,32 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
```
Comment (kerrrusha) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java:23]: `save` is deprecated. please fix at the whole project
```suggestion
            session.persist(movie);
```

**Review by kerrrusha — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java [hunk 5]
```diff
@@ -1,11 +1,27 @@
...
 public class Movie implements Cloneable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String title;
+    @OneToMany
+    @JoinTable(name = "movies_actors",
+            joinColumns = @JoinColumn(name = "movie_id"),
+            inverseJoinColumns = @JoinColumn(name = "actor_id"))
     private List<Actor> actors;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/hibernate/relations/model/Movie.java:25]: Lets check is it really One to many? Is it possible One Actor in many Movies?

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 6]
```diff
@@ -1,16 +1,26 @@
...
-        return null;
+        countryDao.add(country);
+        return country;
     }
 
     @Override
     public Country get(Long id) {
-        return null;
+        return countryDao.get(id).orElseThrow(
+                () -> new DataProcessingException("Cannot get country from DB by ID: " + id));
```
Comment (okuzan) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:24]: dataProcessing is for DAO level prolems, in services better throw EntityNotFoundException

**Review by okuzan — APPROVED** [review]

**Pull request closed without merging**